pkg/pub_package_reader/lib/src/pubspec_content_override.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart' as yaml;
+import 'package:yaml_edit/yaml_edit.dart';
+
+final _logger = Logger('pubspec_content_override');
+
+/// The version regular expression that accepted any character as separator.
+final _lenientRegExp = RegExp(r'(\d+).(\d+).(\d+)' // Version number.
+    r'(-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?' // Pre-release.
+    r'(\+([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?' // Build.
+    );
+
+/// Packages with bad versions:
+const _packagesWithBadVersions = <String>{
+  'assets_audio_player_web',
+  'bloc_pattern',
+  'bokain_models',
+  'built_collection',
+  'built_redux',
+  'built_redux_thunk',
+  'built_value',
+  'built_value_generator',
+  'built_value_test',
+  'built_viewmodel',
+  'built_viewmodel_flutter_built_redux',
+  'built_viewmodel_flutter_redux',
+  'built_viewmodel_generator',
+  'carp_context_package',
+  'codemetrics',
+  'fluttercontactpicker',
+  'interactive_webview',
+  'meta_types',
+  'meta_types_generator',
+  'meta_types_json',
+  'meta_types_json_generator',
+  'meta_types_redux',
+  'meta_types_redux_generator',
+  'mocktail',
+  'mocktail_image_network',
+  'polymer_interop',
+};
+
+/// Override pubspec.yaml content if needed:
+///
+/// - If the archive was created before 2022, we may need to update the versions,
+///   as pub_semver accepted separator characters other than `.`.
+String overridePubspecContentIfNeeded({
+  required String content,
+  required DateTime created,
+}) {
+  // quick checks to skip overrides
+  if (created.year >= 2022) {
+    return content;
+  }
+  if (_packagesWithBadVersions.every((name) => !content.contains(name))) {
+    return content;
+  }
```

Comment:
Why not skip this and leave it for `_tryOverrideContent` to parse YAML and compare the name?

Is this really a performance critical path?

Reply:
removed